src/DynamoCoreWpf/Rendering/HelixRenderPackage.cs

```diff
@@ -555,12 +521,281 @@ public IEnumerable<int> PointIndices
+        public int ColorsStride
+        {
+            get => colorStride;
+            set
+            {
+                if (!AllowLegacyColorOperations)
+                {
+                    throw new LegacyRenderPackageMethodException();
+                }
+                colorStride = value;
+            }
+        }
+
+        #endregion
+
+        #region IRenderLabels implementation
+
+        ///// <summary>
+        ///// Get label data; label string and associated position
+        ///// </summary>
+        public List<Tuple<string, float[]>> LabelData
+        {
+            get
+            {
+                var list = new List<Tuple<string, float[]>>();
+                foreach (var tuple in LabelPlaces)
+                {
+                    list.Add(new Tuple<string, float[]>(tuple.Item1, tuple.Item2.ToArray()));
+                }
+                
+                return list;
+            }
+        }
+
+        /// <summary>
+        /// Add a label position to the render package.
+        /// </summary>
+        public void AddLabel(string label, VertexType vertexType, int index)
+        {
+            Vector3 position;
+            switch (vertexType)
+            {
+                case VertexType.Point:
+                    if (index > points.Positions.Count) {return;}
+                    position = points.Positions[index-1];
+                    break;
+                case VertexType.Line:
+                    if (index > lines.Positions.Count) { return; }
+                    position = lines.Positions[index - 1];
+                    break;
+                case VertexType.Mesh:
+                    if (index > mesh.Positions.Count) { return; }
+                    position = Mesh.Positions[index - 1];
+                    break;
+                default:
+                    return;
+            }
+            LabelPlaces.Add(new Tuple<string, Vector3>(label,position));
+        }
+
+        /// <summary>
+        /// Add a label position to the render package.
+        /// </summary>
+        public void AddLabel(string label, double x, double y, double z)
+        {
+            LabelPlaces.Add(new Tuple<string, Vector3>(label, Vector3ForYUp(x, y, z)));
+        }
+
+        /// <summary>
+        /// A flag indicating whether the render package should auto generate labels
+        /// </summary>
+        public bool AutoGenerateLabels { get; set; } = true;
+
+        /// <summary>
+        /// Clear all label data from the render package.
+        /// </summary>
+        public void ClearLabels()
+        {
+            LabelPlaces.Clear();
+            AutoGenerateLabels = true;
+        }
+
+        #endregion
+
+        #region IRenderPackageSupplement implementation
+
+        /// <summary>
+        /// The number of point vertices colors in the package.
+        /// </summary>
+        public int PointVertexColorCount => points.Colors.Count;
```

Comment:
Why do these implementations differ from those in `DefaultRenderPackage`?

Reply:
Because the helix render package is highly optimized to store data in Helix data structure.  Color, Vertex, etc...   This makes for a faster transfer to the View